common/graphics/persistent_cache.cc

```diff
@@ -291,17 +293,36 @@ bool PersistentCache::IsValid() const {
   return cache_directory_ && cache_directory_->is_valid();
 }
 
-sk_sp<SkData> PersistentCache::LoadFile(const fml::UniqueFD& dir,
-                                        const std::string& file_name) {
+PersistentCache::SkSLCache PersistentCache::LoadFile(
+    const fml::UniqueFD& dir,
+    const std::string& file_name,
+    bool need_key) {
+  SkSLCache result;
   auto file = fml::OpenFileReadOnly(dir, file_name.c_str());
   if (!file.is_valid()) {
-    return nullptr;
+    return result;
   }
   auto mapping = std::make_unique<fml::FileMapping>(file);
-  if (mapping->GetSize() == 0) {
-    return nullptr;
+  if (mapping->GetSize() < sizeof(CacheObjectHeader)) {
+    return result;
```

Comment:
Should we consider some log messages here and below to flag when the file is in a bad state?

Reply:
Added some logs.  Corruption is unlikely because these files are written with `fml::WriteAtomically`, which completes the write and then renames the file if successful.